generate_variant_annotation.py

```diff
@@ -247,56 +122,72 @@ def main():
+        'chrom_b37', 'pos_b37',
+        'ref', 'alt', 'allele_type', 'vep', 'rsid', 'af', 'cadd'
+    ]
 
     # Repartition and write parquet file
     (
-        ht.to_spark(flatten=False)
-          .repartition(out_partitions)
-          .write.parquet(out_parquet)
+        ht
+        .select(*col_order)
+        .to_spark(flatten=False)
+        # .repartition(OUT_PARTITIONS)
+        .coalesce(OUT_PARTITIONS)
+        .write.mode('overwrite').parquet(out_parquet)
+        # .write.format('json').mode('overwrite').option('compression', 'gzip').save(out_parquet.replace('.parquet', '.json.gz'))
     )
 
     # Export site list
-    cols = ['chrom_b37', 'pos_b37', 'chrom_b38', 'pos_b38', 'ref', 'alt', 'rsid']
     (
-        ht.select(*cols)
-          .export(out_sitelist)
+        ht.select(*[
+            'chrom_b37', 'pos_b37',
+            'chrom_b38', 'pos_b38',
+            'ref', 'alt', 'rsid'
+        ])
+        .export(out_sitelist)
     )
 
-    return 0
-
-def parse_population_keys(pop_list):
-    ''' Takes a list of population names and filters to:
-        - Keep gnomad only
-        - Remove male/female sub-stats
-        - Remove _raw (stats before any sample filtering)
-    Params:
-        pop_list (list of str)
-    Returns:
-        Filtered pop_list (list of str)
-    '''
-    pop_filt = []
-    for pop in pop_list:
-        if (
-              pop.startswith('gnomad_') and not
-              pop.endswith('_raw') and not
-              pop.endswith('_male') and not
-              pop.endswith('_female')
-           ):
-           pop_filt.append(pop)
-    return pop_filt
 
+if __name__ == '__main__':
 
-def af_to_maf(af):
-    ''' Convert allele freq to minor allele freq using hail conditional
-    '''
-    return hl.cond(af <= 0.5, af, 1 - af)
+    # Parsing command line arguments
+    parser = argparse.ArgumentParser(
+        description='This script generates variant table for OpenTargets Genetics pipelines.')
+    parser.add_argument('--gnomadFile', help='Hail table of the 3+ version of gnomAD dataset.',
+                        type=str, required=False)
+    parser.add_argument('--chainFile', help='GRCh38 -> GRCh37 liftover chain file.',
+                        type=str, required=False)
+    parser.add_argument('--mafThreshold', help='Lower threshold for minor allele frequency.',
+                        type=float, required=False)
+    parser.add_argument('--outputFolder', help='Directory into which the output files will be saved.',
+                        type=str, required=True)
+    parser.add_argument('--test', help='Number of rows taken for testing and debug purposes',
+                        type=int, required=False)
+
+    args = parser.parse_args()
+
+    gnomad_file = args.gnomadFile if args.gnomadFile else GNOMAD_3_TABLE
+    chain_file = args.chainFile if args.chainFile else CHAIN_FILE
+    maf_threshold = args.mafThreshold if args.mafThreshold else MAF_THRESHOLD
```

Comment:
Suggest rewriting this as argparse default values (as discussed above, they could also be displayed in usage with ArgumentDefaultsHelpFormatter)

Reply:
Yes, good idea.